DR_0011_React-for-paginated-views.md

```diff
@@ -0,0 +1,194 @@
+
+`PaginatedTable` is configurable. Each paginated view has a different API source (where we load the data from), number of columns and template for how each row in the table should be rendered. 
+
+The following properties must be set when initialising `PaginatedTable`:
+
+| Property         | Type              | Description   | 
+| ---------------- | ----------------- | ------------- | 
+| `url`            | `String`          | The API URL where data is loaded from. | 
+| `columns `       | `Array`           | A list of column definitions. The number of elements defines the number of columns in the table. Each element must define the following keys: <br>`label` (the label used in the table head for the column)<br>`field` (the model field, used to apply sorting)<br>`sortable` (identifies if the column is sortable) keys. | 
+| `row_component ` | `React.Component` | A React component that is used to render individual rows into the table. This is necessary because so tables (such as resources) require more complex rendering of individual cells. | 
+
+##### Example use
+
+```html
+<!-- Load React libs -->
+<script crossorigin src="https://unpkg.com/react@15/dist/react.min.js"></script>
+<script crossorigin src="https://unpkg.com/react-dom@15/dist/react-dom.min.js"></script>
+
+<!-- Load the PaginateTable core component -->
+<script src="/static/dist/PaginatedTable.min.js"></script>
+
+<!-- Load the row component used in this view -->
+<script src="/static/dist/PartyRow.min.js"></script>
+<script type="text/javascript">
+  var props = {
+    "url": "/api/v1/resources",
+    "columns": [
+      {"label": "Party", "field": "name", "sortable": true},
+      {"label": "Type", "field": "type", "sortable": true}
+    ],
+    "row_component": Row
+  }
+  ReactDOM.render(
+    React.createElement(PaginatedTable, props),
+    document.getElementById('paginated-table')
+  );
+</script>
+```
+
+A very basic implementation can be found in [branch `react-party`](https://github.com/Cadasta/cadasta-platform/compare/react-party).
+
+### Database changes
+
+N/A
+
+### Security considerations
+
+N/A
+
+### Scalability
+
+When implemented, this approach will improve scalability of the platform because smaller sets will be process in the database and the application server. 
+
+### Maintainability
+
+#### Package management
+
+We will use [yarn](https://yarnpkg.com) for package management. Yarn provides some advantages over npm:
+
+- It is faster.
+- Builds fail less often when a package can't be installed. It retries the failing request instead. 
+- It is deterministic. The same libraries are installed across all systems; with npm you sometimes had different versions of the same library installed depending on the install order and the whether other libraries depend on the same library but a different version. 
+
+#### Task runner
+
+We will use [Grunt](https://gruntjs.com/) as our task runner. Grunt will be used to create builds, run tests and lint the code.  
```

Comment:
Similarly, we should standardize our usage of `grunt` vs `gulp`.

Reply:
I'm not too fussed which one we use, but we should agree on one. I've put an item on the today's meeting agenda

Reply:
👍 agreed on `grunt` in dev team meeting